examples/user-region/src/user_region/core.clj

```diff
@@ -0,0 +1,131 @@
+(ns user-region.core
+  "Demonstrates group-by operations and aggregations on KTable.
+
+  In this specific example we compute the user count per geo-region
+  from a KTable that contains <user, region> information."
+  (:gen-class)
+  (:require [clojure.string :as str]
+            [clojure.pprint :as pp]
+            [clojure.tools.logging :refer [info]]
+            [jackdaw.streams :as j]
+            [jackdaw.serdes.edn :as jse])
+  (:import [org.apache.kafka.common.serialization Serdes]))
+
+(defn app-config
+  "Returns the application config."
+  []
+  {"application.id"            "user-activity"
+   "bootstrap.servers"         "localhost:9092"
+   "cache.max.bytes.buffering" "0"})
+
+(defn topic-names
+  []
+  ["user-region" "region-user-count"])
+
+(defn user-region-entries
+  []
+  (vector ["alice" "asia"]
+          ["bob" "america"]
+          ["chao" "asia"]
+          ["alice" "europe"] ;; Note: Alice moved from Asia to Europe
+          ["eve" "america"]
+          ["fang" "asia"]
+          ["gandalf" "europe"]
+          ["marina" "europe"]))
+
+(defn topic-config
+  ([topic-name]
+   (topic-config topic-name (jse/serde) (jse/serde)))
+  ([topic-name key-serde value-serde]
+   {:topic-name         topic-name
+    :partition-count    1
+    :replication-factor 1
+    :topic-config       {}
+    :key-serde          key-serde
+    :value-serde        value-serde}))
+
+
+(defn topology-builder
+  "Reads from a Kafka topic called `user-region` and writes these to
+  a Kafka topic called `region-user-count`. Returns a topology builder."
+  [builder]
+  (let [user-region-table (j/ktable builder (topic-config "user-region"))
+
+        region-count (-> user-region-table
+                         (j/group-by (fn [[_ v]] (vector v v))
+                                     (topic-config nil
+                                                   (Serdes/String)
+                                                   (Serdes/String)))
+                         (j/count))]
+
+    ;; Aggregate the user counts of by region
```

Comment:
The aggregation happens in an earlier step. This just converts the table to a record stream and publishes to output.